.github/workflows/deploy-html-storybook-prod.yml

```diff
@@ -4,26 +4,73 @@ on:
   push:
     branches:
       - main
+  workflow_dispatch: # Allow manual deployment trigger
+    inputs:
+      deploy_reason:
+        description: "Select deployment type"
+        required: true
+        default: "Manual Trigger"
 
 env:
   AWS_S3_BUCKET_NAME: storybook-html-design-system-blocks-gov-ie
 
 jobs:
+  determine-affected:
+    runs-on: ubuntu-latest
+    outputs:
+      html_affected: ${{ steps.check-affected.outputs.html_affected }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: "20.x"
+
+      - name: Install dependencies
+        run: |
+          corepack enable
+          pnpm install --frozen-lockfile --ignore-scripts
+
+      - name: Determine affected projects
+        id: affected
+        run: |
+          npx nx affected:libs --base=origin/main~1 --head=HEAD > affected.txt
```

Comment:
This command in local doesn't work, does `affected:libs` need to be present in package.json?